sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/SchemaRegistryAvroEncoder.cs

```diff
@@ -306,7 +312,7 @@ private static DatumReader<object> GetReader(Schema writerSchema, Schema readerS
             }
             else
```

Comment:
Should we consider avoiding the array for `recordFormatIdentifier` by using a span in L302-307?

```csharp
ReadOnlySpan<byte> recordFormatIdentifier = default;
if (memory.Length >= RecordFormatIndicatorLength)
{
    recordFormatIdentifier = memory.Slice(0, RecordFormatIndicatorLength).Span
}
if (recordFormatIdentifier != default && recordFormatIdentifier.SequenceEqual(EmptyRecordFormatIndicator))
```

Reply:
ReadOnlySpan cannot be used in async methods.